vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/view/components/ProjectDetailsComponent.java

```diff
@@ -36,13 +41,15 @@ public class ProjectDetailsComponent extends Composite<CardLayout> {
   final TextArea experimentalDesignField = new TextArea();
   final TextArea projectObjective = new TextArea();
 
-  private transient final ProjectDetailsHandler handler;
+  private final transient Handler handler;
 
   public ProjectDetailsComponent(@Autowired ProjectInformationService projectInformationService) {
     Objects.requireNonNull(projectInformationService);
+    this.handler = new Handler(this, projectInformationService);
+    handler.setFieldsEditableOnlyOnFocus();
+    handler.attachSubmissionActionOnBlur();
```

Comment:
I think one of the advantages of having the handler in a seperate class was that the component class does not get so bloated with handler logic. 
These methods should IMO be called during the construction of the handler

Reply:
> I think one of the advantages of having the handler in a separate class was that the component class does not get so bloated with handler logic.
These methods should IMO be called during the construction of the handler

We have the same bloating but even worse outside the context of the component. If we put it as an internal class we still have the separation (you can collapse the class if you do not care about the internals) whilst avoiding clutter in the package.

> Would also allow us to seperate the dependency import necessary for the handler from the ones of the layout 👍

I do not understand your point here, can you elaborate?